From cd5104901f13bbc40c455ba652416a23f2a4756e Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Sat, 27 Nov 2010 17:38:29 +0000
Patch-mainline: 2.6.38
References: fate#310916
Git-commit: 7d2cb39c332146b0906651a8567f8b81af4b1584
Subject: [PATCH 1904/2588] drm/i915: Release fenced GTT mapping on suspend

... so that upon first use after resume we will reacquire the fence reg.

Reported-by: Keith Packard <keithp@keithp.com>
Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
(cherry picked from commit 7d2cb39c332146b0906651a8567f8b81af4b1584)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/i915_gem.c |   11 +++++++++--
 1 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index e8bed54..6539e97 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -1812,8 +1812,15 @@ static void i915_gem_reset_fences(struct drm_device *dev)
 
 	for (i = 0; i < 16; i++) {
 		struct drm_i915_fence_reg *reg = &dev_priv->fence_regs[i];
-		if (reg->obj)
-			i915_gem_clear_fence_reg(reg->obj);
+		struct drm_i915_gem_object *obj = reg->obj;
+
+		if (!obj)
+			continue;
+
+		if (obj->tiling_mode)
+			i915_gem_release_mmap(obj);
+
+		i915_gem_clear_fence_reg(obj);
 	}
 }
 
-- 
1.7.6

